ruskeyverbs/views.py

```diff
@@ -26,19 +51,618 @@ def index(request):
+    (Russian_Sentence, Translation_Text, Quiz_1_Score, Quiz_2_Score,
+    Quiz_3_score, Average_Score )"""
+    request.session['quiz_state'] = 0
+    # quiz state will restrict the user to going through the quizzes in order
+    return render(
+        request,
+        'ruskeyverbs/verb_detail.html',
+        context={'verb': verb,
+                 'example_pk': example_pk,
+                 'not_studied': not_studied}
+    )
+
+
+@login_required
+def StudyPage(request, pk):
+    """
+    page will be displayed before quiz if the user has not yet studied a verb
+    """
+    example_inst = get_object_or_404(Example, pk=pk)
+    russian_text_list = strip_punct_lower(example_inst.russian_text,
+                                          stressed=True).split()
+    """
+    test each word of the sentence to see which one is a form of the verb
+    being studied and determine which form it is
+    """
+    for i in range(len(russian_text_list)):
+        if russian_text_list[i] in example_inst.verb.get_forms_list(stressed=True):
+            target_verb = russian_text_list[i]
+            target_verb_modified = "<b>" + russian_text_list[i] + "</b>"
+    russian_text_bold = example_inst.russian_text.replace(target_verb,
+                                                     target_verb_modified)
+    request.session['quiz-state'] = 0
+    return render(
+        request,
+        'ruskeyverbs/study_session.html',
+        context={
+            'russian_text': russian_text_bold,
+            'english_text': example_inst.translation_text,
+            'pk': pk,
+            'file': example_inst.example_audio,
+        }
+            )
+
+
+@login_required
+def MultipleChoice(request, pk):
+    """
+    view for multiple choice quiz
+    """
+    example_inst = get_object_or_404(Example, pk=pk)
+    russian_text_list = strip_punct_lower(example_inst.russian_text,
+                                          stressed=True).split()
+    unstressed_russian_text_list = strip_punct_lower(
+        example_inst.russian_text).split()
+    """
+    test each word of the sentence to see which one is a form of the verb
+    being studied and determine which form it is
+    """
+    print(request.session)
+    field_names = Verb._meta.get_fields()
+    for i in range(len(russian_text_list)):
+        for field in field_names:
+            try:
+                verb_form = getattr(example_inst.verb, field.name)
+                if verb_form == russian_text_list[i]:
+                    target_field = field
+                    answer = unstressed_russian_text_list[i]
+                    stressed_answer = russian_text_list[i]
+            except AttributeError:
+                pass
+                # pass to get around many to one field returned by get_fields()
+    # create a list of the pks of 3 other random verbs and randomize it
+    choice_pks = [int(example_inst.verb.pk)]
+    while len(choice_pks) < 4:
```

Comment:
This could probably be done more efficiently. And I tend to try to avoid unbounded while loops as if for some reason you don't hit your 'exit' condition your app will just hang forever.

Reply:
Swapped in a random.choices() method here.

Reply:
Good